### DISCUSSION TITLE
Adding custom video-based cutscenes

### DISCUSSION THREAD

**Discussion by CooperJ-0** [KadeArchive/Kade-Engine#411] [Q&A] [answered]
Howdy y'all,

I've been at this all day, trying to figure out how in the blazes mods like Whitty and 'Smoke em out struggle' are able to pull off video-based cutscenes. And I've finally been directed here. 

Now I'm going to be honest here and say I haven't looked at the documentation for the kade engine all too well, and I'm not quite certain if there is any.  I can't read for the life of me, so I tend to try and find video resources covering the topics I'm trying to learn. I did watch the video describing how to make a mod in three hours (gj, btw), but that didn't seem to include anything on video cut scenes. Are there any? If not, understandable, but where is the documentation for the kade engine? 

If there is none, and no example code from either of those mods, am I wrong to guess that it was done by creating an object that's asset file is just the frames of animation, with an accompanying XML file? Or are there supportive video-playing methods in the haxeflixel library?

Any and all help would be greatly, greatly appreciated. Thanks!

**Comment by AtonyGit** [comment]
I don't know exactly, but my best idea is to use code from the week 6 cutscene (it might be in PlayState.hx where the transition to thorns occurs)

**Reply by CooperJ-0** [reply]
It looks like that just uses an object and flashes though several frames of it's animation over and over again

**Reply by CooperJ-0** [reply]
which is a valid way to go about this, but I don't know if  I can make a png large enough to have all the frames of a 30 fps animation play out for like 20 seconds, times 3 songs

**Reply by AtonyGit** [reply]
true, but there's not really a limit to the amount of images

**Reply by CooperJ-0** [reply]
That's also a very valid solution, thank you so much!

**Comment by Kade-github** [comment, marked as answer]
Whitty and Smoke Em' out Struggle both use the technique of creating a sprite sheet which is animated in adobe flash (adobe animate), and then playing it in the game.

**Reply by CooperJ-0** [reply]
is it played in the game using the same animation calls for the sprites?

**Reply by CooperJ-0** [reply]
if so, wouldn't the png of all the frames be of ridiculous dimension?

**Reply by CooperJ-0** [reply]
nvm I'm dumb apparently png has a max size of like 2,500 megapixels, so that's not really a concern.

Thank you so so much for the help!

**Reply by puyoxyz** [reply]
I don't think Kade meant, like, the whole screen being an animation, but like, the characters being animated?
